pandas/core/dtypes/dtypes.py

```diff
@@ -1087,7 +1090,7 @@ def na_value(self) -> NaTType:
 
     def __eq__(self, other: object) -> bool:
         if isinstance(other, str):
-            return other in [self.name, capitalize_first_letter(self.name)]
+            return other in {self.name, capitalize_first_letter(self.name)}
```

Comment:
Sorry, I'm still a bit confused.

First thing is that `self.name` is always going to be `period[...` in lower case, this is the implementation (just two methods above this part):

```python
    @property
    def name(self) -> str_type:
        return f"period[{self._freqstr}]"
```

Then, if we want to compare strings, why not simply:

```suggestion
            return other in {self.name, capitalize_first_letter(self.name)}
            return other[0].lower() + other[1:] == self.name
```

This will make:

```python
assert PeriodDtype('3D') == 'Period[3D]'
assert PeriodDtype('3D') == 'period[3D]'
assert PeriodDtype('3D') != 'Period[3d]'
```

Which for what understand is what we want, no?

Then, there is another case to consider:

```python
>>> PeriodDtype(freq='1s1ms') == PeriodDtype(freq='1001ms')
True
>>> PeriodDtype(freq='1s1ms') == '1001ms'
False
```

Is this what we want? I personally think this is a bug. I'd say that a better implementation here is simply:

```suggestion
            other = self.__class__(other)
```

So the comparison with string is consistent with the comparison with the types directly. If we want to do this, we probably don't want to raise when the string is not a valid frequency string, so I guess this makes more sense:

```suggestion
            try:
                other = self.__class__(other)
            except ValueError:
                return False
```

If we go with this option, we probably should add a test, as this would be a bug fix, not a clean up anymore.

What do you think? I think it's a quick change, but if you prefer that I open a new PR for this I can do it.

Reply:
Maybe we can use pd.PeriodDtype(freq='1s1ms').freq.freqstrt to make the comparison
``` 
>>> pd.PeriodDtype(freq='1s1ms') == pd.PeriodDtype(freq='1001ms')
True
>>> pd.PeriodDtype(freq='1s1ms').freq.freqstr == '1001ms'
True
```

Reply:
I updated this PR to use `return other[:1].lower() + other[1:] == self.name`

For the `PeriodDtype(freq='1s1ms') == '1001ms'`, I would prefer not to make a call about this case in this PR. My gut would prefer to be stricter in eq and require `period[...]`, while just the frequency string can be compared exactly to  `freqstr` as @natmokval suggested. For example, `1001ms` could be "timedelta-like" for a user which shouldn't be made equal to a "period-like`"